leaderboard/llvm_codesize/random_search/README.md

```diff
@@ -0,0 +1,130 @@
+# Random Search
+
+**tldr;**
+A pure random policy that records the best result found within a fixed time
+budget.
+
+**Authors:**
+Facebook AI Research
+
+**Results:**
+
+|  | Walltime (mean) | Codesize Reduction (geomean) |
+| --- | --- | --- |
+| [Random search, p=1.25, t=10](results_p125_t10.csv) | 42.939s | 1.031× |
+| [Random search, p=1.25, t=60](results_p125_t60.csv) | 91.215s | 1.045× |
+| [Random search, p=1.25, t=3600](results_p125_t3600.csv) | 3,630.821s | 1.061× |
+| [Random search, p=1.25, t=10800](results_p125_t10800.csv) | 10,512.356s | 1.062× |
+
+
+**Publication:**
+<!-- TODO(cummins): Add CompilerGym citation when ready. -->
+
+**CompilerGym version:**
+0.1.4
+
+**Open source?**
+Yes, MIT licensed. [Source Code](random_search.py).
+
+**Did you modify the CompilerGym source code?**
+No.
+
+**What parameters does the approach have?**
+Search time *t*, patience *p* (see "Description" below).
+
+**What range of values were considered for the above parameters?** Search time
+is fixed and is indicated by the leaderboard entry name, e.g. "Random search
+(t=30)" means a random search for a minimum 30 seconds. Eight values for
+patience were considered, *n/4*, *n/2*, *3n/4*, *n*, *5n/4*, *3n/2*, *7n/8*, and
+*2n*, where *n* is the size of the initial action space. The patience value was
+selected using the `blas-v0` dataset for validation, see appendix below.
+
+**Is the policy deterministic?**
+No.
+
+## Description
+
+This approach uses a simple random agent on the action space of the target
+program. This is equivalent to running `python -m
+compiler_gym.bin.random_search` on the programs in the test set.
+
+The random search operates by selecting actions randomly until a fixed number of
+steps (the "patience" of the search) have been evaluated without an improvement
+to reward. The search stops after a predetermined amount of search time has
+elapsed.
+
+Pseudo-code for this search is:
+
+```c++
+float search_with_patience(CompilerEnv env, int patience, int search_time) {
```

Comment:
Why is the pseudo-code in C++? Is there a C++ interface too?

Reply:
No particular reason, it just feels more pseudo-y to me with types :)

We don't advertise a C++ interface, but it is entirely usable as a C++ library. Here's is an example random search. It looks pretty much the same as in python:

https://github.com/facebookresearch/CompilerGym/blob/development/examples/RandomSearch.cc#L107-L136